src/mlpack/core/optimizers/scd/descent_policies/greedy_descent.hpp

```diff
@@ -0,0 +1,55 @@
+/**
+ * @file greedy_descent.hpp
+ * @author Shikhar Bhardwaj
+ *
+ * Greedy descent policy for Stochastic Co ordinate Descent (SCD).
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_SCD_DESCENT_POLICIES_GREEDY_HPP
+#define MLPACK_CORE_OPTIMIZERS_SCD_DESCENT_POLICIES_GREEDY_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Greedy descent policy for Stochastic Co-ordinate Descent(SCD). This
+ * descent scheme picks a the co-ordinate for the descent with the maximum
+ * guaranteed descent.
+ */
+class GreedyDescent
+{
+ public:
+  /**
+   * The DescentFeature method is used to get the descent coordinate for the
+   * current iteration.
+   *
+   * @tparam ResolvableFunctionType The type of the function to be optimized.
+   * @param numEpoch The iteration number for which the feature is to be
+   *    obtained.
+   * @param iterate The current value of the decision variable.
+   * @param function The function to be optimized.
+   * @return The index of the coordinate to be descended.
+   */
+
+  // TODO: Find a way to implement this.
```

Comment:
I am planning to implement this descent policy which gives the feature with the maximum guaranteed descent, based on : http://www.maths.ed.ac.uk/~prichtar/Optimization_and_Big_Data_2015/slides/Schmidt.pdf and ideas from this implementation of SCD : http://ttic.uchicago.edu/~tewari/code/scd/

The above mentioned implementation uses a parameter "rho", which is a part of the objective function, which is 0.25 for logistic and hinge loss and 1 for quadratic loss in the part for calculating the predicted descent and finding the feature with the steepest descent.